src/cljs/athens/views/athena.cljs

```diff
@@ -205,10 +214,24 @@
       ;; TODO: change scroll as user reaches top or bottom
       ;; TODO: what happens when user goes to -1? or past end of list?
       (= key KeyCodes.UP)
-      (swap! state update :index dec)
+      (do
+        (swap! state update :index #(dec (if (zero? index) (count results) index)))
+        (let [cur-index (:index @state)
+              input-el (.. e -target)
+              result-el (.. (. input-el closest "div.athena") -lastElementChild)
+              next-el (nth (array-seq (.. result-el -children)) cur-index)]
+          (when (is-beyond-rect? next-el result-el)
+            (.. next-el (scrollIntoView (not= cur-index (dec (count results))) {:behavior "auto"})))))
```

Comment:
After a long day, trying to read this is requiring a lot of effort, so I think there's too much going on.